### ISSUE TITLE
Stripe payments not working - Mandatory configuration key STRIPE_CONNECTED_ID not present

### ISSUE THREAD

**Issue by cougil** [alfio-event/alf.io#392]
Hi.
We are using a PaaS where we have deployed a new version of the db that is using Alf.io. We've found that something wrong happened.  
We've checked that now payments with Stripe are not working:
```
 Caused by: java.lang.IllegalArgumentException: Mandatory configuration key STRIPE_CONNECTED_ID not present
 	at alfio.manager.system.ConfigurationManager.lambda$getRequiredValue$2(ConfigurationManager.java:142) ~[classes!/:?]
 	at java.util.Optional.orElseThrow(Optional.java:290) ~[?:1.8.0_161-heroku]
 	at alfio.manager.system.ConfigurationManager.getRequiredValue(ConfigurationManager.java:142) ~[classes!/:?]
 	at alfio.manager.system.ConfigurationManager$$FastClassBySpringCGLIB$$7630228.invoke(<generated>) ~[classes!/:?]
 	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204) ~[spring-core-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:738) ~[spring-aop-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157) ~[spring-aop-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
 	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:99) ~[spring-tx-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
 	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:282) ~[spring-tx-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
 	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:96) ~[spring-tx-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179) ~[spring-aop-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
 	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:673) ~[spring-aop-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
 	at alfio.manager.system.ConfigurationManager$$EnhancerBySpringCGLIB$$fa35437d.getRequiredValue(<generated>) ~[classes!/:?]
 	at alfio.manager.StripeManager.options(StripeManager.java:212) ~[classes!/:?]
 	at alfio.manager.StripeManager.chargeCreditCard(StripeManager.java:191) ~[classes!/:?]
 	at alfio.manager.PaymentManager.processStripePayment(PaymentManager.java:85) ~[classes!/:?]
 	... 100 more
 ERROR alfio.controller.ControllerExceptionHandler - unexpected exception
org.springframework.transaction.UnexpectedRollbackException: Transaction rolled back because it has been marked as rollback-only
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.commit(AbstractPlatformTransactionManager.java:724) ~[spring-tx-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
	at org.springframework.transaction.interceptor.TransactionAspectSupport.commitTransactionAfterReturning(TransactionAspectSupport.java:504) ~[spring-tx-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:292) ~[spring-tx-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:96) ~[spring-tx-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179) ~[spring-aop-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:673) ~[spring-aop-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
	at alfio.manager.TicketReservationManager$$EnhancerBySpringCGLIB$$f3e6d6.confirm(<generated>) ~[classes!/:?]
	at alfio.controller.ReservationController.handleReservation(ReservationController.java:469) ~[classes!/:?]
20
```
Please, have a look, how we can solve it? We are about to open the sales!

**Comment by cougil** [comment]
I've solved the issue just inserting a new row in the 'Configuration' table containing a new entry `STRIPE_CONNECTED_ID` that has the same value as `STRIPE_CONNECT_CLIENT_ID` , so I suppose there is something wrong this constant ...

**Comment by cbellone** [comment]
please check if you have enable "Platform Mode". If yes, disable it.

![image](https://user-images.githubusercontent.com/3385346/36350039-7b3996f8-1490-11e8-9271-9d418b5ed0aa.png)

Platform mode uses [Stripe Connect](https://stripe.com/connect) for multi-tenant environments, where the manager of the instance collects fees from (other) event organizers for each ticket sold. I don't think it could be useful for your use case.

If you want to collect fees from customers, you can add a new Additional Option with the following policy:

![image](https://user-images.githubusercontent.com/3385346/36350065-2021c7c6-1491-11e8-86ea-4741569aa8ea.png)

**Comment by cougil** [comment]
Hi.
I've checked and in the system & event the "Platform Mode" is disabled while at the organization it was not changed "Leave unchanged".
Thanks for the tip about the fees for customers! Didn't notice about how to manage it!

**Comment by cbellone** [comment]
this should be solved now. Will be released in the upcoming v1.14
